events/app.py

```diff
@@ -27,7 +27,18 @@ def get_all_events():
 @app.route('/v1/search', methods=['GET'])
 def search_event():
     """Search for the event with the given name in the DB."""
-    pass
+    try:
+        event_name = request.args['name']
+        events = app.config["COLLECTION"].find({'name': event_name})
+
+        events = [Event(**ev).dict for ev in events]
+        events_dict = build_events_dict(events)
```

Comment:
Why can't this just take Event objects, rather than Event objects rendered as dictionaries?

Reply:
In particular, you have the same dictionary comprehension on line 19 as here.